### PR TITLE
adrv9009: Add iio application.

### PR DESCRIPTION
**Pull request by cristiantpop**
This instantiates "iio_axi_adc" and "iio_axi_dac" modules, that permits
reading/writing and parametrization, of "axi_adc" and "axi_dac" devices.

Signed-off-by: Cristian Pop <cristian.pop@analog.com>

### CODE REVIEW

**Comment by cristiantpop** [comment]
V2:

- Fix codacy;

**Comment by cristiantpop** [comment]
V3:

- Remove duplicated "#include "uart.h"".

File: projects/adrv9009/src/app/headless.c [hunk 1]
```diff
@@ -47,6 +47,29 @@
...
 
+#ifdef IIO_EXAMPLE
+#include "irq.h"
+#include "irq_extra.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "tinyiiod.h"
+#include "iio_axi_adc.h"
+#include "iio_axi_dac.h"
+#include "iio.h"
```
Comment (mirceacaprioru) [inline projects/adrv9009/src/app/headless.c]: nitpick: this should be sorted

File: projects/adrv9009/src/app/headless.c [hunk 2]
```diff
@@ -870,6 +893,184 @@ int main(void)
...
+	status = iio_axi_dac_init(&iio_axi_dac_inst, &iio_axi_dac_init_par);
+	if(status < 0)
+		return status;
+
+	iio_axi_adc_intf_par = (struct iio_interface_init_par) {
+		.dev_name = iio_axi_adc_inst->adc->name,
+		.dev_instance = iio_axi_adc_inst,
+		.iio_device = iio_axi_adc_create_device(iio_axi_adc_inst->adc->name,
+							iio_axi_adc_inst->adc->num_channels),
+			      .get_xml = iio_axi_adc_get_xml,
```
Comment (mirceacaprioru) [inline projects/adrv9009/src/app/headless.c]: the indent here seems a bit off

File: projects/adrv9009/src/app/headless.c [hunk 3]
```diff
@@ -870,6 +893,184 @@ int main(void)
...
+			      .read_data = iio_axi_adc_read_dev,
+			      .write_data = NULL,
+	};
+
+	iio_axi_dac_intf_par = (struct iio_interface_init_par) {
+		.dev_name = iio_axi_dac_inst->dac->name,
+		.dev_instance = iio_axi_dac_inst,
+		.iio_device = iio_axi_dac_create_device(iio_axi_dac_inst->dac->name,
+							iio_axi_dac_inst->dac->num_channels),
+			      .get_xml = iio_axi_dac_get_xml,
```
Comment (mirceacaprioru) [inline projects/adrv9009/src/app/headless.c]: same here

File: projects/adrv9009/src/devices/adi_hal/parameters.h [hunk 4]
```diff
@@ -137,6 +137,10 @@
 #define TX_DMA_BASEADDR			XPAR_AXI_ADRV9009_TX_DMA_BASEADDR
 
 #define DDR_MEM_BASEADDR		XPAR_DDR_MEM_BASEADDR
+
+#define UART_DEVICE_ID			XPAR_XUARTPS_0_DEVICE_ID
+#define UART_IRQ_ID			XPAR_XUARTPS_1_INTR
```
Comment (adrimbarean) [inline projects/adrv9009/src/devices/adi_hal/parameters.h]: Is this the correct interrupt ID for UART device 0? I was expecting it to be `XPAR_XUARTPS_0_INTR`.

**Comment by cristiantpop** [comment]
V4:

- Fix indentation.

- Sort headers.

**Comment by cristiantpop** [comment]
If there are no other comments here, I will rebase it later today.

**Comment by cristiantpop** [comment]
IIO_EXAMPLE application, should be placed in another file, since is platform independent.

**Pull request closed without merging**